### PR TITLE
Rails Forms: Working with Form_tag and Form_for methods

### PR DESCRIPTION
**Pull request by ldelbel**
In this project, my partner and I worked on building forms in the views of a generic User model, including the following methods: 

- New
- Create
- Edit
- Update

We used rubocop to ensure code structure quality.

### CODE REVIEW

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great start @team :clap: However, some requirements need to be implemented in order to get approval for this project. Check :arrow_down: 

#### UsersController

##### `update` action

* [x] `redirect_to edit_user_path` if user was saved
* [x] `render :edit` if user was not saved ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_update.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

##### private `user_params` method

* [ ] requires correct fields, i.e. `params.require(:user).permit(:username, :email, :password)` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_params.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

#### Views

##### `new`

###### commented out `form`

* [x] uses `users_path` helper for action
* [x] uses `post` for method
* [x] uses `"UTF-8"` for ccept-charset
* [x] uses `user[<field_name>]` for input's name
* [x] uses form_authenticity_token

```html
<form action=users_path method='post' accept-charset="UTF-8">
  <label for='user[username]'>Username</label>
  <input name='user[username]' type='text'>
  <input type="hidden" name="authenticity_token" value="<%= form_authenticity_token %>">
  <input type="submit" value="Submit">
</form>
```

([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form1.png) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

###### commented out `form_tag`

* [x] uses `label_tag`, `text_field_tag`, `email_field_tag`, `password_field_tag`

```html
<%= form_tag(users_path) do %>
  <%= label_tag 'username' %>
  <%= text_field_tag 'username' %>
  <%= label_tag 'email' %>
  <%= email_field_tag 'email' %>
  <%= label_tag 'password' %>
  <%= password_field_tag 'password' %>
  <%= submit_tag %>
<% end %>
```

([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_tag2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

###### `form_for`

* [x] uses `@user` from controller

```html
<%= form_for @user do |u|%>
  <%= u.label :username %>
  <%= u.text_field :username %>
  <%= u.label :email %>
  <%= u.email_field :email %>
  <%= u.label :password %>
  <%= u.password_field :password %>
  <%= u.submit %>
<% end %>
```
### That's it :relieved: When finished, kindly submit a code re-review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

**Comment by ldelbel** [comment]
Hey @stratospherique ! I appreciate the review. It was very useful and instructive.

I did all the changes but the change requested for private user_params method because the function is already present in users_helper.rb. I ask for the next TSE to kindly address that and confirm if it is suitable.

**Review by davidmrtz-dev — CHANGES_REQUESTED** [review]
## Hi Team  👨‍🚀 
### Very good job so far implementing the requested changes.

However, there is one implementation that needs to be addressed before moving on.  👨‍💻  ⬇️ 
Please check out below.

****
- [x] Users controller. => requires correct fields, i.e. params.require(:user).permit(:username, :email, :password) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_params.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
As pointed out above, its required to pass the form parameters in a safe way, 
Please add the private method that prevent this using "params.require"
Consider the following example:
```ruby
private

  def user_params
    params.require(:user).permit(:username, :email, :password,
                                 :password_confirmation)
  end
```

Responding to your request, in this case, it is preferred to have this method inside the controller since helpers, although can be used to create methods related to the controller, they are commonly used to implement methods that handle some type of logically related to views, to involve all the logic in the helper and just call the methods in your views.
In this project, it is not so noticeable, but in large projects, it becomes a convention to have the methods that constitute the intrinsic logic of the controller within the controller.
I hope I have clarified your doubt

****

### **Please make the changes required and submit for re-review once you are ready!** 

****

### Status: Changes required. ♻️  
****

Happy coding!  💻  
Have a great day 🙂

If you need more clarifications kindly reach me on slack.
**[TSE:  David Elí]** 
Slack: @ David Eli Martinez Garcia

File: app/controllers/users_controller.rb
Comment (mcrd25) [inline app/controllers/users_controller.rb]: Please remove this commented out code before merging.

**Review by mcrd25 — APPROVED** [review]
# Changed Approved :heavy_check_mark: 

Hi Team! :wave: 

Thanks for successfully implementing changes requested from previous review! :+1: 

You may now move on! :checkered_flag: 

**Please remove any commented out code before merging!**

***If you have any questions, now or in the future, do not hesitate to contact me on social media, email or on slack and I will be glad to chat with you.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

**Comment by ElijahTobs** [comment]
@mcrd25 Thanks for the review.

The commented out code has been deleted.

**Pull request merged**
